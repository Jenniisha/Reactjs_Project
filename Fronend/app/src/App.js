import { Route, Routes, Router } from "react-router";
import './App.css';
import Home from './components/Home';

import DoctorContainer from "./components/DoctorContainer";
import store from "./store/myStore";
import { Provider } from 'react-redux';
import NoPageFound from "./components/NoPageFound";
import AddDoctorContainer from "./components/AddDoctorContainer";




function App() {
  return (
    <Provider store={store}>

      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/doctors/add" element={<AddDoctorContainer mode="ADD" />} />
        <Route path="/doctors/edit/" element={<AddDoctorContainer mode="EDIT" />} />
        <Route path="/doctors/search/speciality/" element={<DoctorContainer />} />
        <Route path="*" element={<NoPageFound />} />
      </Routes>


    </Provider>



  );
};




export default App;