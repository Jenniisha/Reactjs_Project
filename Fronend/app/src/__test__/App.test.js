import App from '../App';
import {shallow, configure } from 'enzyme';
import tojson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });


test('two plus two is four', () => {
  expect(2 + 2).toBe(4);
});


describe('test app renders or not', () => {
  test('Chck app comes without any issues', () => {
    shallow(<App />);
  });
});

describe('Basic App rendering test cases 1', () => {
  test('Taking snapshot', () => {
    const tree = shallow(<App />);
    expect(tojson(tree)).toMatchSnapshot();
  });

});