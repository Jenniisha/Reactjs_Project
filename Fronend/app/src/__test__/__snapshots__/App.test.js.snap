// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic App rendering test cases 1 Taking snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Routes>
    <Route
      element={<Home />}
      path="/"
    />
    <Route
      element={
        <Memo(Connect(AddDoctor))
          mode="ADD"
        />
      }
      path="/doctors/add"
    />
    <Route
      element={
        <Memo(Connect(AddDoctor))
          mode="EDIT"
        />
      }
      path="/doctors/edit/"
    />
    <Route
      element={<Memo(Connect(SearchDoctor)) />}
      path="/doctors/search/speciality/"
    />
    <Route
      element={<NoPageFound />}
      path="*"
    />
  </Routes>
</Provider>
`;
